lib/src/core/sheba/methods.dart

```diff
@@ -0,0 +1,59 @@
+import '../../constants/sheba/constants.dart';
+
+/// Takes Sheba code and gives information from it
+class Sheba {
+  final String _shebaCode;
+
+  /// [RegExp] to checking validation of the sheba
+  final pattern = RegExp(r'IR[0-9]{24}');
+
+  /// [RegExp] to finding bank code
+  final patternCode = RegExp(r'IR[0-9]{2}([0-9]{3})[0-9]{19}');
+
+  Sheba(this._shebaCode);
```

Comment:
it's better to move the class constructor to the top of the class. and make it `const` constructor if it's possible.

Reply:
the `RegExp` class hasn't `const` constructor and we can't use it.